pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+# PyScript Core Documentation
+
+ * [Terminology](#terminology) - what we mean by "_term_" in this document
+ * [Bootstrapping core](#bootstrapping-core) - how to enable PyScript Next in your page
+ * [How Scripts Work](#how-scripts-work) - how `<script type="...">` works
+ * [How Events Work](#how-events-work) - how `<button py-click="...">` works
+ * [XWorker](#xworker) - how `XWorker` class and its `xworker` reference work
+ * [Custom Scripts](#custom-scripts) - how *custom types* can be defined and used to enrich any core feature
+
+
+## Terminology
+
+This section goal is to avoid confusion around topics discussed in this document, describing each *term* as exhaustively as possible.
+
+<details>
+  <summary><strong>Interpreter</strong></summary>
+  <div>
+
+Also commonly referred as *runtime* or *engine*, we consider an **interpreter** any "_piece of software_" able to parse, understand, and ultimately execute, a *Programming Language* through this project.
+
+We also explicitly use that "_piece of software_" as the interpreter name it refers to. We currently bundle references to four interpreters:
+
+ * [pyodide](https://pyodide.org/en/stable/index.html) is the name of the interpreter that runs likely the most complete version of latest *Python*, enabling dozen official modules at run time, also offering a great *JS* integration in its core
+ * [micropython](https://micropython.org/) is the name of the interpreter that runs a small subset of the *Python* standard library and is optimized to run in constrained environments such as *Mobile* phones, or even *Desktop*, thanks to its tiny size and an extremely fast bootstrap
+ * [wasmoon](https://github.com/ceifa/wasmoon) is the name of the interpreter that runs *Lua* on the browser and that, among the previous two interpreters, is fully compatible with all core features
+ * [ruby-wasm-wasi](https://github.com/ruby/ruby.wasm) is the name of the (currently *experimental*) interpreter that adds *Ruby* to the list of programming languages currently supported
```

Comment:
Isn't wasmoon also "experimental"? Or do you mean the interpreter itself is experimental?

Reply:
wasmoon can do everything Pyodide or MicroPython can do ... why do you think it's experimental?